### PR TITLE
Add confirmation modal component and general modal component

### PR DESCRIPTION
**Pull request by bodichka11**
Screenshot 2023-08-21 150804
Screenshot 2023-08-21 151436
Screenshot 2023-08-21 151454
Screenshot 2023-08-21 151615
Screenshot 2023-08-21 151513
Screenshot 2023-08-21 151528
Screenshot 2023-08-21 151543
Screenshot 2023-08-21 151633

### CODE REVIEW

File: frontend/src/app/shared/components/confirmation-modal/confirmation-modal.component.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+
+@Component({
+    selector: 'confirmation-modal',
+    templateUrl: './confirmation-modal.component.html',
+})
+export class ConfirmationModalComponent implements OnInit {
+    constructor(public activeModal: NgbActiveModal) {}
+
+    @Input() buttons: any[] = [];
```
Comment (tatianahutii) [inline frontend/src/app/shared/components/confirmation-modal/confirmation-modal.component.ts:11]: please remove any

File: frontend/src/app/shared/components/confirmation-modal/confirmation-modal.component.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    @Input() buttonClickHandler: (text: string) => void;
+
+    @Output() newItemEvent = new EventEmitter<NgbActiveModal>();
```
Comment (tatianahutii) [inline frontend/src/app/shared/components/confirmation-modal/confirmation-modal.component.ts:14]: why do we need these?

File: frontend/src/app/shared/components/general-modal/general-modal.component.html [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+</div>
+<div class="modal-footer">
+    <div class="row">
+        <div class="col" *ngFor="let but of buttons">
+            <ng-container *ngIf="isButtonVisible">
+                <button type="button" class="btn btn-outline-dark m-1" (click)="but.handler()">{{but.text}}</button>
+            </ng-container>
+        </div>
+    </div>
+</div>
```
Comment (tatianahutii) [inline frontend/src/app/shared/components/general-modal/general-modal.component.html:16]: should not be hard-coded component. Component should be passed as a parammeter

File: frontend/src/app/shared/components/general-modal/general-modal.component.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    styleUrls: ['./general-modal.component.sass'],
+})
+export class GeneralModalComponent implements OnInit {
+    @Input() isButtonVisible: boolean;
+    constructor(public activeModal: NgbActiveModal) {}
+
+    @Input() buttons: any[] = [];
+    @Input() buttonClickHandler: (text: string) => void;
+
+    @Output() newItemEvent = new EventEmitter<NgbActiveModal>();
```
Comment (tatianahutii) [inline frontend/src/app/shared/components/general-modal/general-modal.component.ts:16]: see the previous comments

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Review by tatianahutii — APPROVED** [review]

**Review by Den313 — DISMISSED** [review]

**Review by tatianahutii — APPROVED** [review]

**Pull request merged**